alphabase/psm_reader/msfragger_reader.py

```diff
@@ -32,8 +32,12 @@ def _get_mods_from_masses(sequence, msf_aa_mods):
         _mass_str, site_str = mod.split('@')
         mod_mass = float(_mass_str)
         site = int(site_str)
+        cterm_position = len(sequence) + 1
         if site > 0:
-            mod_mass = mod_mass - AA_ASCII_MASS[ord(sequence[site-1])]
+            if site < cterm_position:
+                mod_mass = mod_mass - AA_ASCII_MASS[ord(sequence[site-1])]
+            else:
+                mod_mass -= (MASS_H + MASS_O + MASS_PROTON)
```

Comment:
Removing a proton does not make sense here, as only neutral masses are involved

Reply:
I double checked my results with this file:
[20190131_QExHFX3_Ogris_MFPL_gel_PP2A_EV_90p.zip](https://github.com/MannLabs/alphabase/files/10970413/20190131_QExHFX3_Ogris_MFPL_gel_PP2A_EV_90p.zip)

With the current code from the pull request, for the peptide RTPDYFL with Methyl@cterm, we get the following fragment masses, where both b ions (without the cterm mod) and y ions (with the cterm mod) match the calculated fragment masses from http://db.systemsbiology.net/proteomicsToolkit/FragIonServlet?sequence=RTPDYFL&massType=monoRB&charge=1&bCB=1&yCB=1&nterm=0&cterm=14.015650&addModifType=&addModifVal=

```
b_z1           y_z1           b_z2           y_z2  
157.108387     769.376683     79.057832      385.191980
258.156066     668.329005     129.581671     334.668140
```

This is also present in the MSFragger source code, where an extra mass of proton is present when reporting mass of cterm mod, but not nterm. This is also evident in the pepxml entry
```
<search_hit peptide="RTPDYFL" massdiff="0.00225830078125" calc_neutral_pep_mass="924.47046" peptide_next_aa="-" num_missed_cleavages="1" num_tol_term="2" protein_descr="Serine/threonine-protein phosphatase 2A catalytic subunit beta isoform OS=Mus musculus OX=10090 GN=Ppp2cb PE=1 SV=1" num_tot_proteins="2" tot_num_ions="12" hit_rank="1" num_matched_ions="7" protein="sp|P62715|PP2AB_MOUSE" peptide_prev_aa="R" is_rejected="0">
<alternative_protein protein_descr="Serine/threonine-protein phosphatase 2A catalytic subunit alpha isoform OS=Mus musculus OX=10090 GN=Ppp2ca PE=1 SV=1" protein="sp|P63330|PP2AA_MOUSE" peptide_prev_aa="R" peptide_next_aa="-" num_tol_term="2"/>
<modification_info mod_cterm_mass="32.025665" modified_peptide="RTPDYFLc[32]">
</modification_info>
```
where the cterm_mass is the sum of methylation mass (14.02 Da) + the masses of hydrogen, oxygen, and proton.

Can I provide any other tests to show that the mass of the proton should be included here?

Reply:
I will try it by myself. If this is indeed the case, I will merge this PR. 

PS: it is still wired for me, removing a proton will retain the electron, leading to a negative charge...

Reply:
Hi @yangkl96 , i think this is a H2O, not HO+proton, please double check this and then I will merge this PR, thanks.

Reply:
Thanks for the explanation @jalew188. I talked to Fengchao and we agree that it was weird that our pepxml writer was using proton mass instead of another hydrogen mass. We may change this in the future MSFragger code, but it should work fine now since the ppm difference is so small. I have pushed the correction.

Reply:
Yes, i agree. Just to make sure the substractor is correct, and not confused